Docs: No basic role documentation updates

linoman opened the pull request:
**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

linoman committed:
WIP

linoman committed:
Merge remote-tracking branch 'origin/main' into linoman/no_basic_role_revisit_documentation

linoman committed:
Add more scenarios to documentation

linoman committed:
Merge remote-tracking branch 'origin/main' into linoman/no_basic_role_revisit_documentation

linoman committed:
Address CI comments

linoman committed:
Update documentation

linoman committed:
Spell check

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/manage-rbac-roles/index.md:
```diff
@@ -257,6 +258,10 @@ If the default basic role definitions do not meet your requirements, you can cha
 - Determine the permissions you want to add or remove from a basic role. For more information about the permissions associated with basic roles, refer to [RBAC role definitions]({{< relref "./rbac-fixed-basic-role-definitions/#basic-role-assignments" >}}).
 
+{{% admonition type="note" %}}
+You cannot add permissions to the `No Basic Role` basic role.
```
I like the wording of the admonition you wrote a bit further down. Does that work for you?

gamab replied:
The basic role none has no permission and cannot be modified.

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -31,7 +31,7 @@ Your rollout strategy should help you answer the following questions:
 ## Review basic role and fixed role definitions
 
-As a first step in determining your permissions rollout strategy, we recommend that you become familiar with basic role and fixed role definitions. In addition to assigning fixed roles to any user and team, you can also modify basic roles permissions, which changes what a Viewer, Editor, or Admin can do. This flexibility means that there are many combinations of role assignments for you to consid ...
+As a first step in determining your permissions rollout strategy, we recommend that you become familiar with basic role and fixed role definitions. In addition to assigning fixed roles to any user and team, you can also modify basic roles permissions, which changes what a Viewer, Editor, or Admin can do. This flexibility means that there are many combinations of role assignments for you to consid ...
```
```suggestion
As a first step in determining your permissions rollout strategy, we recommend that you become familiar with basic role and fixed role definitions. In addition to assigning fixed roles to any user and team, you can also modify basic roles permissions, which changes what a Viewer, Editor, or Admin can do. This flexibility means that there are many combinations of role assignments for you to consider. If you have a large number of Grafana users and teams, we recommend that you make a list of which fixed roles you might want to use. Keep in mind that `No Basic Role`, which is a role without permissions, cannot be modified or updated.
```

gamab replied:
I don't understand what you are trying to say @linoman.
Is it something along the line:
```suggestion
As a first step in determining your permissions rollout strategy, we recommend that you become familiar with basic role and fixed role definitions. In addition to assigning fixed roles to any user and team, you can also modify basic roles permissions, which changes what a Viewer, Editor, or Admin can do. This flexibility means that there are many combinations of role assignments for you to consider. If you have a large number of Grafana users and teams, we recommend that you make a list of which fixed roles you might want to use. Keep in mind that you can use the `No Basic Role`, which is a basic role without permissions that cannot be modified or updated.
```

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+In the scenario where you want users to have granted access by the team they belong to, we recommend to set users role to `No Basic Role` and let the team assignment assign the role instead.
+
+1. In Grafana, ensure the following configuration settings are enabled.
+   1. `users.auto_assign_org = true` this will automatically assign users to the default organization.
```
How do these substeps render in the published docs? If they render as `1.`, I recommend manually changing these substeps to a., b., c. etc.

linoman replied:
I tried to render them with letters but the style was not the same as the numbers. I refactored it in order to show a configuration example instead.

chri2547 replied:
Looks good! Thank you!

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+   1. `users.auto_assign_org_id = <org_id>` this will automatically assign users to the specified organization.
+   1. `users.auto_assign_org_role = None` this will automatically assign users to the specified role.
+   1. Restart the Grafana instance.
+2. Create a team with the desired name.
```
these can be formatted as 1.

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+3. Set the fixed roles needed for the team.
+4. Add users to the team.
+
+This way, a user will be added automatically to the default organization and have no permissions until it has been assigned to a team.
```
```suggestion
A user will be added to the default organization automatically but won't have any permissions until assigned to a team.
```

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+### Allow user to access only to a specific dashboard
+
+In the scenario where you want to allow a user access to a specific dashboard, we recommend the `No Basic Role` as a starting point for the user and then assign the dashboard permissions to the user.
+However, in order to access the dashboard more permissions are needed other than just the dashboard. Here's a list of permissions needed in order to access a dashboard:
```
```suggestion
Below is a list of permissions required to access a dashboard:
```

chri2547 replied:
I removed the "However..." sentence because it seems like we address that in line 326. Just checking that this makes sense to you and is ok.

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+In the scenario where you want to allow a user access to a specific dashboard, we recommend the `No Basic Role` as a starting point for the user and then assign the dashboard permissions to the user.
+However, in order to access the dashboard more permissions are needed other than just the dashboard. Here's a list of permissions needed in order to access a dashboard:
+
+1. The dashboard owner must give permission to the user
```
```suggestion
1. The dashboard owner must give permission to the user.
```

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+
+1. The dashboard owner must give permission to the user
+1. The admin must grant the following permissions to the user
+   1. annotations:read
```
Do these permission need to be assigned in the order you have listed? If not, you can use a bulleted list here.

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+   1. annotations:read
+   1. datasource:read
+
+The permissions list here are the bare minimum needed to be able to view the dashboard. However, keep in mind that if the dashboard has multiple features, the permissions needed to see the dashboard might be more than just the ones listed here.
```
```suggestion
Please note that the permissions listed below are the minimum required to access the dashboard. It's important to keep in mind that if the dashboard has multiple features, additional permissions may be required beyond those listed here.
```

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+
+The permissions list here are the bare minimum needed to be able to view the dashboard. However, keep in mind that if the dashboard has multiple features, the permissions needed to see the dashboard might be more than just the ones listed here.
+
+### Reducing scope of service accounts
```
```suggestion
### Reduce scope of service accounts
```

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+
+### Reducing scope of service accounts
+
+Service Accounts are great tools for M2M communications. However, they are also a security risk if not properly scoped. To reduce the scope of a service account, you can create a Service Account with `No Basic Role` as a starting point and then assign the permissions needed for the service account.
```
```suggestion
Using Service Accounts is an efficient way to facilitate M2M communications. However, they can pose a security threat if not scoped appropriately. To limit the scope of a service account, you can begin by creating a Service Account with "No Basic Role" and then assign the necessary permissions for the account.
```

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+
+Service Accounts are great tools for M2M communications. However, they are also a security risk if not properly scoped. To reduce the scope of a service account, you can create a Service Account with `No Basic Role` as a starting point and then assign the permissions needed for the service account.
+
+1. Head over to the Service Accounts page and add a new Service Account.
```
```suggestion
1. Refer to [Service Accounts]() and add a new Service Account.
```

chri2547 replied:
Add a link to the doc.

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+
+1. Head over to the Service Accounts page and add a new Service Account.
+1. Set the necessary roles for the Service Account.
+   1. Set the basic role to `No Basic Role`.
```
Same question here about how these substeps render. If they render as `1.`, then manually change these to a. and b.

linoman replied:
Same as before, I didn't like the final rendered style so I refactored it in order to keep some visual coherence.

chri2547 commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+   1. Set the basic role to `No Basic Role`.
+   1. Set the fixed roles needed for the Service Account.
+
+This will reduce the permissions needed for the Service Account and reduce the risk of the Service Account being compromised.
```
```suggestion
This will reduce the required permissions for the Service Account and minimize the risk of compromise.
```

chri2547 reviewed (commented):
Some suggestions and comments. I'd like another review after you take a look at my feedback. Really nice addition to the docs, thank you!

gamab commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+
+### Manage user permissions through teams
+
+In the scenario where you want users to have granted access by the team they belong to, we recommend to set users role to `No Basic Role` and let the team assignment assign the role instead.
```
```suggestion
In the scenario where you want users to have access granted by the team they belong to, we recommend to set users role to `No Basic Role` and let the team assignment assign the role instead.
```

gamab commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+   1. `users.auto_assign_org_role = None` this will automatically assign users to the specified role.
+   1. Restart the Grafana instance.
+2. Create a team with the desired name.
+3. Set the fixed roles needed for the team.
```
```suggestion
3. Assigned fixed roles to the team.
```

gamab commented on docs/sources/administration/roles-and-permissions/access-control/plan-rbac-rollout-strategy/index.md:
```diff
@@ -301,3 +305,41 @@ roles:
+
+This way, a user will be added automatically to the default organization and have no permissions until it has been assigned to a team.
+
+### Allow user to access only to a specific dashboard
```
I don't think I agree with this scenario and would not mention it here, at least yet.
I strongly feel like "No Basic Role" is an incorrect answer to the problem: "Viewers can access all dashboards by default".
IMHO, this use case is actually a feature request to: "Allow modifying the default dashboard/folder permissions".
Wdyt?

gamab commented on docs/sources/administration/roles-and-permissions/access-control/rbac-fixed-basic-role-definitions/index.md:
```diff
@@ -159,9 +160,10 @@ This will allow you to control access to different OnCall features using the fol
+| Admin         | `plugins:grafana-oncall-app:admin`  | Default [Grafana organization administrator]({{< relref "../#basic-roles" >}}) assignments.             |
+| Editor        | `plugins:grafana-oncall-app:editor` | Default [Editor]({{< relref "../#basic-roles" >}}) assignments.                                         |
+| Viewer        | `plugins:grafana-oncall-app:reader` | Default [Viewer]({{< relref "../#basic-roles" >}}) assignments.                                         |
+| None          | `plugins:grafana-oncall-app:none`   | Default [None]({{< relref "../#basic-roles" >}}) assignments.                                           |
```
I don't think OnCall define the role none yet.
```suggestion
```

linoman replied:
Should OnCall have the role defined?

gamab commented on docs/sources/setup-grafana/configure-security/planning-iam-strategy/index.md:
```diff
@@ -149,7 +149,7 @@ You can assign roles through the user interface or APIs, establish them through
 ### What are roles?
 
-Within an organization, Grafana has established three primary [organization roles]({{< relref "../../../administration/roles-and-permissions#organization-roles" >}}) - organization administrator, editor, and viewer - which dictate the user's level of access and permissions, including the ability to edit data sources or create teams.
+Within an organization, Grafana has established three primary [organization roles]({{< relref "../../../administration/roles-and-permissions#organization-roles" >}}) - organization administrator, editor, and viewer - which dictate the user's level of access and permissions, including the ability to edit data sources or create teams. Additionally, Grafana has an empty role that can be used as a st ...
```
When I read this I understand that you can modify the none basic role. I would write:
```suggestion
Within an organization, Grafana has established three primary [organization roles]({{< relref "../../../administration/roles-and-permissions#organization-roles" >}}) - organization administrator, editor, and viewer - which dictate the user's level of access and permissions, including the ability to edit data sources or create teams. Grafana also has [an empty basic role]({{<insert link here>}}) that grants no permission to the user. With this role users will solely inherit permissions from their teams or directly assigned roles.
```

gamab reviewed (commented):
Overall great job, I left a couple of comments that we can go over together if you want :)

linoman committed:
Apply suggestions from code review

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

linoman committed:
Address PR comments

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

linoman committed:
Remove documentation updates until we reach further clarification

linoman committed:
Address PR comments

linoman commented:
I have addressed all comments from @gamab and @chri2547. Thank you for your review!

However, there are 2 comments that are out of scope from this PR.

1. There are 2 solutions in order to have private dashboards. Either remove permissions from the viewer role or assign users the role none and build it up from there.
2. The oncall plugin might need to be updated in order to work with the new role.

chri2547 reviewed (approved):
Looks great! Thanks for addressing my comments!

linoman committed:
Merge remote-tracking branch 'origin/main' into linoman/no_basic_role_revisit_documentation
